molgenis-data-rest/src/main/java/org/molgenis/data/rest/v2/RestControllerV2.java

```diff
@@ -299,7 +323,52 @@ public EntityCollectionBatchCreateResponseBodyV2 createEntities(@PathVariable("e
 			response.setStatus(HttpServletResponse.SC_NO_CONTENT);
 			throw e;
 		}
+	}
+
+	/**
+	 * Copy an entity.
+	 * 
+	 * @param entityName
+	 *            name of the entity that will be copied.
+	 * @param request
+	 *            CopyEntityRequestV2
+	 * @param response
+	 *            HttpServletResponse
+	 * @return String name of the new entity
+	 * @throws Exception
+	 */
+	@RequestMapping(value = "copy/{entityName}", method = POST, produces = APPLICATION_JSON_VALUE)
```

Comment:
You're creating a new resource when you copy a repository, so a PUT(or POST?) to the location where that resource is going to reside would be a bit more RESTful perhaps?